algoliasearch-core/src/test/java/com/algolia/search/JacksonParserTest.java

```diff
@@ -1004,4 +1004,11 @@ void recommendations() throws JsonProcessingException {
     assertThat(recommendHit.getObjectID()).isEqualTo("D05927-8161-111");
     assertThat(recommendHit.getScore()).isEqualTo(32.72f);
   }
+
+  @Test
+  void consequenceParams_emptyQuery_nullEdits() throws JsonProcessingException {
+    ConsequenceParams params = new ConsequenceParams().setQuery("");
+    String json = Defaults.getObjectMapper().writeValueAsString(params);
+    assertThat(json).isEqualTo("{\"query\":\"\"}");
```

Comment:
Does the engine accept this value? Asking because I am not sure :)

Reply:
I've made some tests and it seems to accept empty query :)

![Screenshot 2022-01-06 at 11 53 05 AM](https://user-images.githubusercontent.com/1970868/148371990-cbb84e72-bc9a-4713-b646-a359ec35698a.png)